Homework_2.playground/Contents.swift

```diff
@@ -1,47 +1,115 @@
-class Stack {
-    private let linkedList = LinkedList()
+/// Container for elements organized with LIFO principle.
+class Stack<Element> {
+    private let linkedList = LinkedList<Element>()
     
-    func push(element: Int) {
+    /// Adds a new element on top of the stack
+    ///
+    ///     let stack = Stack<Int>
+    ///     stack.push(element: 7)
+    ///     //top -> 7
+    ///
+    /// - Parameter element: The element to push on top of the stack
+    /// - Complexity: O(1).
+    func push(element: Element) {
         linkedList.append(element)
     }
     
-    func pop() -> Int? {
+    /// Removes and returns the first element on top of the stack
+    ///
+    ///     let stack = Stack<Int>()
+    ///     stack.push(element: 7)
+    ///     stack.push(element: 9)
+    ///     //top -> 9 -> 7
+    ///     print(stack.pop())
+    ///     //prints "9"
+    ///     //top -> 7
+    ///
+    /// - Returns: The first element on top of the stack, or nil if stack is empty
+    /// - Complexity: O(1)
+    func pop() -> Element? {
         return linkedList.removeFirst()
     }
     
-    func top() -> Int? {
+    /// Returns the first element on top of the stack
+    ///
+    ///     let stack = Stack<Int>()
+    ///     stack.push(element: 7)
+    ///     stack.push(element: 9)
+    ///     print(stack.top())
+    ///     //prints "9"
+    ///     //top -> 9 -> 7
+    ///
+    /// - Returns: The first element on top of the stack, or nil if stack is empty
+    /// - Complexity: O(1)
+    func top() -> Element? {
         return linkedList.first
     }
 }
 
-class LinkedList {
-    private var head: Frame?
-    var first: Int? {
+/// Linear collection of data elements, where each element points to the next element.
+class LinkedList<Element> {
+    private var head: Frame<Element>?
+    
+    /// Returns the first element in the list, or nil if list is empty
+    var first: Element? {
         return head?.data
     }
     
-    func append(_ element: Int) {
+    /// Adds a new element at the head of the list
```

Comment:
What is "at the head of the list"?

Also I would expect to see details about how this method affects `first` property.

Reply:
I changed `head` to` top` for clarity. Added details about affects to the `first` property.

Reply:
Ok, now what is "at the top of the list"?

Reply:
>I changed head to top for clarity. Added details about affects to the first property.

Do you think you have to add the same note for `removeFirst()` function?

Reply:
Check the last change, I think I understood what you meant.